lib/ansible/plugins/action/__init__.py

```diff
@@ -734,6 +734,30 @@ def _execute_module(self, module_name=None, module_args=None, tmp=None, task_var
 
         self._update_module_args(module_name, module_args, task_vars)
 
+        # FIXME: convert async_wrapper.py to not rely on environment variables
+        # make sure we get the right async_dir variable, backwards compatibility
+        # means we need to lookup the env value ANSIBLE_ASYNC_DIR first
+        remove_async_dir = None
+        if wrap_async or self._task.async_val:
+            env_async_dir = [e for e in self._task.environment if
```

Comment:
Corner case, but isn't this an immediately breaking change for anyone that uses `ANSIBLE_ASYNC_DIR` and uses `async` on Windows hosts today (since with this change, Windows hosts will now use the env value which is probably not a Windows path)? /me shakes fist at env config for stuff like this

Hate to say it, but we may need to consult an ignore flag on the shell plugin for this so it won't pick up the env value at all for powershell...

Reply:
This is only if someone has set the following on the task

```
environment:
  ANSIBLE_ASYNC_DIR: /path
```

This wouldn't make any sense today so I don't think it would break anybodies playbooks. Because there's no environment config value set on the powershell plugin then people who have set `ANSIBLE_ASYNC_DIR` already won't affect PowerShell.

This is all moot really as only the deprecated format works today on the POSIX side.